src/NHibernate.Test/SystemTransactions/TransactionNotificationFixture.cs

```diff
@@ -182,7 +182,6 @@ public void ShouldNotifyAfterDistributedTransactionWithOwnConnection(bool doComm
 			{
 				using (var tx = new TransactionScope())
 				{
-					ownConnection1.EnlistTransaction(System.Transactions.Transaction.Current);
```

Comment:
This was undue. The enlistment is done by NHibernate even on provided connections. (When I have worked on the system transaction mess (see [NH-4011](https://nhibernate.jira.com/browse/NH-4011)), I have initially considered not handling transaction enlistment on provided connections, but changed my mind later and forgot about this line here.)

Due to `HanaConnection` not supporting being asked to enlist in a transaction in which it is already enlisted, the test was failing for HANA.